src/ralph/api/routers/statements.py

```diff
@@ -416,19 +423,21 @@ async def put(
     _enrich_statement_with_authority(statement_as_dict, current_user)
 
     try:
-        existing_statement = DATABASE_CLIENT.query_statements_by_ids([statementId])
+        existing_statements = desync(
+            BACKEND_CLIENT.query_statements_by_ids([statementId])
+        )
     except BackendException as error:
         raise HTTPException(
             status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
             detail="xAPI statements query failed",
         ) from error
 
-    if existing_statement:
+    if existing_statements:
         # The LRS specification calls for deep comparison of duplicate statement ids.
         # In the case that the current statement is not equivalent to one found
         # in the database we return a 409, otherwise the usual 204.
-        for existing in existing_statement:
-            if not statements_are_equivalent(statement_as_dict, existing["_source"]):
+        async for existing in existing_statements:
+            if not statements_are_equivalent(statement_as_dict, existing):
```

Comment:
Just to make sure, was `_source` was removed somewhere else ? I couldn't find the origin of this change...

Reply:
We don't wrap statements in a document with `{"_id": <id>, "_source": <statement>}` anymore with the new unified data backends